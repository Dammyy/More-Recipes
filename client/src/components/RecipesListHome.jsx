import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import RecipeHome from './RecipeHome';
import ActionButtons from './HomeActionButtons';
/**
 *
 *
 * @class RecipeListHome
 * @extends {PureComponent}
 */
class RecipesListHome extends PureComponent {
  /**
   *
   *
   * @returns {void}
   *
   * @memberOf RecipeList
   */
  render() {
    const {
      recipes, toggleModal, firstName, logout
    } = this.props;
    return (
      <div>
        <div className="col-md-12 home-logged-in">
          <ActionButtons logout={logout} firstName={firstName} />
        </div>
        <div className="container scrollable">
          <div className="row">
            {
          recipes
            .map((recipe, i) => {
              return (
                <RecipeHome
                  {...recipe}
                  key={recipe.id}
                  i={i}
                  toggleModal={toggleModal}
                />
              );
            })
        }
          </div>
          <hr />
        </div>
      </div>
    );
  }
}
RecipesListHome.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.any).isRequired,
  toggleModal: PropTypes.func.isRequired,
  i: PropTypes.number.isRequired,
  firstName: PropTypes.string.isRequired,
  logout: PropTypes.func.isRequired,
};
export default RecipesListHome;
