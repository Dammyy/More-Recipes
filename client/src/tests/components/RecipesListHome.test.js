import React from 'react';
import { shallow } from 'enzyme';
import RecipesListHome from '../../components/RecipesListHome';

describe('RecipesListHome component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        recipes: {
          recipes: [
            {
              id: 47,
              title: 'fgrgrgr',
              ingredients: 'sdgewrgreg',
              details: 'rgergergreg',
              image: 'https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU',
              reviews: 0,
              upvotes: 0,
              downvotes: 0,
              favorited: 0,
              createdAt: '2018-03-17T13:19:08.125Z',
              updatedAt: '2018-03-17T13:19:08.125Z',
              userId: 27
            }
          ],
        }
      };
      const component = shallow(<RecipesListHome {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
