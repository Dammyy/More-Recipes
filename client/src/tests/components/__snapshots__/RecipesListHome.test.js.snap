// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipesListHome
    recipes={
      Object {
        "recipes": Array [
          Object {
            "createdAt": "2018-03-17T13:19:08.125Z",
            "details": "rgergergreg",
            "downvotes": 0,
            "favorited": 0,
            "id": 47,
            "image": "https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU",
            "ingredients": "sdgewrgreg",
            "reviews": 0,
            "title": "fgrgrgr",
            "updatedAt": "2018-03-17T13:19:08.125Z",
            "upvotes": 0,
            "userId": 27,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="latest-h3"
        >
          Latest Recipes
        </h3>,
        <div
          className="col-md-12 latest-recipes"
        >
          <div
            className="row"
          >
            <RecipeHome
              createdAt="2018-03-17T13:19:08.125Z"
              details="rgergergreg"
              downvotes={0}
              favorited={0}
              id={47}
              image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
              ingredients="sdgewrgreg"
              reviews={0}
              title="fgrgrgr"
              updatedAt="2018-03-17T13:19:08.125Z"
              upvotes={0}
              userId={27}
            />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Latest Recipes",
          "className": "latest-h3",
        },
        "ref": null,
        "rendered": "Latest Recipes",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <RecipeHome
              createdAt="2018-03-17T13:19:08.125Z"
              details="rgergergreg"
              downvotes={0}
              favorited={0}
              id={47}
              image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
              ingredients="sdgewrgreg"
              reviews={0}
              title="fgrgrgr"
              updatedAt="2018-03-17T13:19:08.125Z"
              upvotes={0}
              userId={27}
            />
          </div>,
          "className": "col-md-12 latest-recipes",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RecipeHome
                createdAt="2018-03-17T13:19:08.125Z"
                details="rgergergreg"
                downvotes={0}
                favorited={0}
                id={47}
                image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
                ingredients="sdgewrgreg"
                reviews={0}
                title="fgrgrgr"
                updatedAt="2018-03-17T13:19:08.125Z"
                upvotes={0}
                userId={27}
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "47",
              "nodeType": "class",
              "props": Object {
                "createdAt": "2018-03-17T13:19:08.125Z",
                "details": "rgergergreg",
                "downvotes": 0,
                "favorited": 0,
                "id": 47,
                "image": "https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU",
                "ingredients": "sdgewrgreg",
                "reviews": 0,
                "title": "fgrgrgr",
                "updatedAt": "2018-03-17T13:19:08.125Z",
                "upvotes": 0,
                "userId": 27,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="latest-h3"
          >
            Latest Recipes
          </h3>,
          <div
            className="col-md-12 latest-recipes"
          >
            <div
              className="row"
            >
              <RecipeHome
                createdAt="2018-03-17T13:19:08.125Z"
                details="rgergergreg"
                downvotes={0}
                favorited={0}
                id={47}
                image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
                ingredients="sdgewrgreg"
                reviews={0}
                title="fgrgrgr"
                updatedAt="2018-03-17T13:19:08.125Z"
                upvotes={0}
                userId={27}
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Latest Recipes",
            "className": "latest-h3",
          },
          "ref": null,
          "rendered": "Latest Recipes",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <RecipeHome
                createdAt="2018-03-17T13:19:08.125Z"
                details="rgergergreg"
                downvotes={0}
                favorited={0}
                id={47}
                image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
                ingredients="sdgewrgreg"
                reviews={0}
                title="fgrgrgr"
                updatedAt="2018-03-17T13:19:08.125Z"
                upvotes={0}
                userId={27}
              />
            </div>,
            "className": "col-md-12 latest-recipes",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <RecipeHome
                  createdAt="2018-03-17T13:19:08.125Z"
                  details="rgergergreg"
                  downvotes={0}
                  favorited={0}
                  id={47}
                  image="https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU"
                  ingredients="sdgewrgreg"
                  reviews={0}
                  title="fgrgrgr"
                  updatedAt="2018-03-17T13:19:08.125Z"
                  upvotes={0}
                  userId={27}
                />,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "47",
                "nodeType": "class",
                "props": Object {
                  "createdAt": "2018-03-17T13:19:08.125Z",
                  "details": "rgergergreg",
                  "downvotes": 0,
                  "favorited": 0,
                  "id": 47,
                  "image": "https://cdn.filestackcontent.com/fP7AK7PSNWpcFQJojqTU",
                  "ingredients": "sdgewrgreg",
                  "reviews": 0,
                  "title": "fgrgrgr",
                  "updatedAt": "2018-03-17T13:19:08.125Z",
                  "upvotes": 0,
                  "userId": 27,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
